### ISSUE TITLE
conda env export does not include pip installed libraries with pip 18.0

### ISSUE THREAD

**Issue by dennisobrien** [conda/conda#7581]
**I'm submitting a...**
  - [x] bug report
  - [ ] feature request

## Current Behavior

After upgrading an environment to pip 18.0, `conda env export` no longer includes a section of pip installed libraries.

### Steps to Reproduce
Create a new conda environment.
```
dobrien@dobrien-ubuntu:~$ conda create --name=tempenv python=3.6 --yes
Fetching package metadata .............
Solving package specifications: .

Package plan for installation in environment /home/dobrien/anaconda3/envs/tempenv:

The following NEW packages will be INSTALLED:

    ca-certificates: 2018.4.16-0       conda-forge
    certifi:         2018.4.16-py36_0  conda-forge
    libgcc-ng:       7.2.0-hdf63c60_3             
    ncurses:         5.9-10            conda-forge
    openssl:         1.0.2o-0          conda-forge
    pip:             18.0-py36_0       conda-forge
    python:          3.6.5-1           conda-forge
    readline:        7.0-0             conda-forge
    setuptools:      40.0.0-py36_0     conda-forge
    sqlite:          3.20.1-2          conda-forge
    tk:              8.6.8-0           conda-forge
    wheel:           0.31.1-py36_0     conda-forge
    xz:              5.2.3-0           conda-forge
    zlib:            1.2.11-h470a237_3 conda-forge

ca-certificate 100% |###################################################################################| Time: 0:00:01 136.18 kB/s
tk-8.6.8-0.tar 100% |###################################################################################| Time: 0:00:07 419.66 kB/s
openssl-1.0.2o 100% |###################################################################################| Time: 0:00:01   1.89 MB/s
zlib-1.2.11-h4 100% |###################################################################################| Time: 0:00:00 343.36 kB/s
python-3.6.5-1 100% |###################################################################################| Time: 0:00:04   5.24 MB/s
setuptools-40. 100% |###################################################################################| Time: 0:00:00   1.14 MB/s
wheel-0.31.1-p 100% |###################################################################################| Time: 0:00:00 451.89 kB/s
pip-18.0-py36_ 100% |###################################################################################| Time: 0:00:00   2.42 MB/s
#
# To activate this environment, use:
# > source activate tempenv
#
# To deactivate an active environment, use:
# > source deactivate
#
```

Activate this environment and export the env.  Note that it installs pip 18.0 (released 3 days ago).
```
dobrien@dobrien-ubuntu:~$ source activate tempenv
(tempenv) dobrien@dobrien-ubuntu:~$ conda env export
name: tempenv
channels:
- conda-forge
- defaults
dependencies:
- ca-certificates=2018.4.16=0
- certifi=2018.4.16=py36_0
- ncurses=5.9=10
- openssl=1.0.2o=0
- pip=18.0=py36_0
- python=3.6.5=1
- readline=7.0=0
- setuptools=40.0.0=py36_0
- sqlite=3.20.1=2
- tk=8.6.8=0
- wheel=0.31.1=py36_0
- xz=5.2.3=0
- zlib=1.2.11=h470a237_3
- libgcc-ng=7.2.0=hdf63c60_3
prefix: /home/dobrien/anaconda3/envs/tempenv
```

Install something with pip.
```
(tempenv) dobrien@dobrien-ubuntu:~$ pip install hyperopt
Collecting hyperopt
Collecting nose (from hyperopt)
  Downloading https://files.pythonhosted.org/packages/15/d8/dd071918c040f50fa1cf80da16423af51ff8ce4a0f2399b7bf8de45ac3d9/nose-1.3.7-py3-none-any.whl (154kB)
    100% |████████████████████████████████| 163kB 4.6MB/s 
Collecting six (from hyperopt)
  Downloading https://files.pythonhosted.org/packages/67/4b/141a581104b1f6397bfa78ac9d43d8ad29a7ca43ea90a2d863fe3056e86a/six-1.11.0-py2.py3-none-any.whl
Collecting numpy (from hyperopt)
  Downloading https://files.pythonhosted.org/packages/88/29/f4c845648ed23264e986cdc5fbab5f8eace1be5e62144ef69ccc7189461d/numpy-1.15.0-cp36-cp36m-manylinux1_x86_64.whl (13.9MB)
    100% |████████████████████████████████| 13.9MB 1.6MB/s 
Collecting future (from hyperopt)
Collecting networkx (from hyperopt)
  Downloading https://files.pythonhosted.org/packages/11/42/f951cc6838a4dff6ce57211c4d7f8444809ccbe2134179950301e5c4c83c/networkx-2.1.zip (1.6MB)
    100% |████████████████████████████████| 1.6MB 10.0MB/s 
Collecting scipy (from hyperopt)
  Downloading https://files.pythonhosted.org/packages/a8/0b/f163da98d3a01b3e0ef1cab8dd2123c34aee2bafbb1c5bffa354cc8a1730/scipy-1.1.0-cp36-cp36m-manylinux1_x86_64.whl (31.2MB)
    100% |████████████████████████████████| 31.2MB 845kB/s 
Collecting pymongo (from hyperopt)
  Downloading https://files.pythonhosted.org/packages/11/88/dd1f8c4281a60791b043f55e338d0521049208f21e3de19ddc9c160dbbef/pymongo-3.7.1-cp36-cp36m-manylinux1_x86_64.whl (405kB)
    100% |████████████████████████████████| 409kB 14.1MB/s 
Collecting decorator>=4.1.0 (from networkx->hyperopt)
  Downloading https://files.pythonhosted.org/packages/bc/bb/a24838832ba35baf52f32ab1a49b906b5f82fb7c76b2f6a7e35e140bac30/decorator-4.3.0-py2.py3-none-any.whl
Building wheels for collected packages: networkx
  Running setup.py bdist_wheel for networkx ... done
  Stored in directory: /home/dobrien/.cache/pip/wheels/44/c0/34/6f98693a554301bdb405f8d65d95bbcd3e50180cbfdd98a94e
Successfully built networkx
Installing collected packages: nose, six, numpy, future, decorator, networkx, scipy, pymongo, hyperopt
Successfully installed decorator-4.3.0 future-0.16.0 hyperopt-0.1 networkx-2.1 nose-1.3.7 numpy-1.15.0 pymongo-3.7.1 scipy-1.1.0 six-1.11.0
```
Export the env.  Notice the library we pip installed is not listed.
```
(tempenv) dobrien@dobrien-ubuntu:~$ conda env export
name: tempenv
channels:
- conda-forge
- defaults
dependencies:
- ca-certificates=2018.4.16=0
- certifi=2018.4.16=py36_0
- ncurses=5.9=10
- openssl=1.0.2o=0
- pip=18.0=py36_0
- python=3.6.5=1
- readline=7.0=0
- setuptools=40.0.0=py36_0
- sqlite=3.20.1=2
- tk=8.6.8=0
- wheel=0.31.1=py36_0
- xz=5.2.3=0
- zlib=1.2.11=h470a237_3
- libgcc-ng=7.2.0=hdf63c60_3
prefix: /home/dobrien/anaconda3/envs/tempenv
```

It is listed with `pip freeze`.
```
(tempenv) dobrien@dobrien-ubuntu:~$ pip freeze
certifi==2018.4.16
decorator==4.3.0
future==0.16.0
hyperopt==0.1
networkx==2.1
nose==1.3.7
numpy==1.15.0
pymongo==3.7.1
scipy==1.1.0
six==1.11.0
```

Downgrading the pip==10.0.1 solves the problem.
```
(tempenv) dobrien@dobrien-ubuntu:~$ conda install pip==10.0.1
Fetching package metadata .............
Solving package specifications: .

Package plan for installation in environment /home/dobrien/anaconda3/envs/tempenv:

The following packages will be DOWNGRADED:

    pip: 18.0-py36_0 conda-forge --> 10.0.1-py36_0 conda-forge

Proceed ([y]/n)? y

pip-10.0.1-py3 100% |###################################################################################| Time: 0:00:00   2.01 MB/s
(tempenv) dobrien@dobrien-ubuntu:~$ conda env export
DEPRECATION: The legacy format has been deprecated and will be removed in the future.
name: tempenv
channels:
- conda-forge
- defaults
dependencies:
- ca-certificates=2018.4.16=0
- certifi=2018.4.16=py36_0
- ncurses=5.9=10
- openssl=1.0.2o=0
- pip=10.0.1=py36_0
- python=3.6.5=1
- readline=7.0=0
- setuptools=40.0.0=py36_0
- sqlite=3.20.1=2
- tk=8.6.8=0
- wheel=0.31.1=py36_0
- xz=5.2.3=0
- zlib=1.2.11=h470a237_3
- libgcc-ng=7.2.0=hdf63c60_3
- pip:
  - decorator==4.3.0
  - future==0.16.0
  - hyperopt==0.1
  - networkx==2.1
  - nose==1.3.7
  - numpy==1.15.0
  - pymongo==3.7.1
  - scipy==1.1.0
  - six==1.11.0
prefix: /home/dobrien/anaconda3/envs/tempenv

```

## Expected Behavior

`conda env export` should include pip installed modules when using pip==18.0.

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
Current conda install:

               platform : linux-64
          conda version : 4.3.29
       conda is private : False
      conda-env version : 4.3.29
    conda-build version : 1.21.3
         python version : 3.5.3.final.0
       requests version : 2.14.2
       root environment : /home/dobrien/anaconda3  (writable)
    default environment : /home/dobrien/anaconda3/envs/tempenv
       envs directories : /home/dobrien/anaconda3/envs
                          /home/dobrien/.conda/envs
          package cache : /home/dobrien/anaconda3/pkgs
                          /home/dobrien/.conda/pkgs
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.continuum.io/pkgs/main/linux-64
                          https://repo.continuum.io/pkgs/main/noarch
                          https://repo.continuum.io/pkgs/free/linux-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/linux-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/linux-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : /home/dobrien/.condarc
             netrc file : /home/dobrien/.netrc
           offline mode : False
             user-agent : conda/4.3.29 requests/2.14.2 CPython/3.5.3 Linux/4.4.0-130-generic debian/stretch/sid glibc/2.23    
                UID:GID : 1000:1000
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
==> /home/dobrien/.condarc <==
channels:
  - conda-forge
  - defaults
ssl_verify: True
```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
# packages in environment at /home/dobrien/anaconda3/envs/tempenv:
#
ca-certificates           2018.4.16                     0    conda-forge
certifi                   2018.4.16                py36_0    conda-forge
decorator                 4.3.0                     <pip>
future                    0.16.0                    <pip>
hyperopt                  0.1                       <pip>
libgcc-ng                 7.2.0                hdf63c60_3    defaults
ncurses                   5.9                          10    conda-forge
networkx                  2.1                       <pip>
nose                      1.3.7                     <pip>
numpy                     1.15.0                    <pip>
openssl                   1.0.2o                        0    conda-forge
pip                       18.0                     py36_0    conda-forge
pymongo                   3.7.1                     <pip>
python                    3.6.5                         1    conda-forge
readline                  7.0                           0    conda-forge
scipy                     1.1.0                     <pip>
setuptools                40.0.0                   py36_0    conda-forge
six                       1.11.0                    <pip>
sqlite                    3.20.1                        2    conda-forge
tk                        8.6.8                         0    conda-forge
wheel                     0.31.1                   py36_0    conda-forge
xz                        5.2.3                         0    conda-forge
zlib                      1.2.11               h470a237_3    conda-forge
```
</p></details>

**Comment by jamesmyatt** [comment]
This affects me too

**Comment by kalefranz** [comment]
Looks like this will be resolved by https://github.com/conda/conda/pull/7612

**Comment by albertmichaelj** [comment]
I’m still seeing this issue as well with conda 4.6.1 and pip 18.1 (I also tried with pip 10.0.1, but that didn’t work as well). With conda 4.6.1, I can’t seem to export the pip packages at all.

**Comment by msarahan** [comment]
fixed in 4.6.2 (available on conda-canary channel right now).  please see https://github.com/conda/conda/issues/8136
